src/SourceBuild/content/eng/tools/BinaryToolKit/DetectBinaries.cs

```diff
@@ -43,6 +36,41 @@ public static async Task<List<string>> ExecuteAsync(string targetDirectory)
         return binaryFiles;
     }
 
+    private static async Task<List<string>> GetIgnoredPatterns(string targetDirectory)
+    {
+        string gitDirectory = Path.Combine(targetDirectory, ".git");
+        bool isGitRepo = Directory.Exists(gitDirectory);
+        if (!isGitRepo)
+        {
+            // Configure a fake git repo to use so that we can run git clean -ndx
+            await ExecuteProcessAsync("git", $"-C {targetDirectory} init -q");
+        }
+
+        string output = await ExecuteProcessAsync("git", $"-C {targetDirectory} clean -ndx");
+
+        List<string> ignoredPaths = output.Split(Environment.NewLine)
+            .Select(line => GitCleanRegex.Match(line))
+            .Where(match => match.Success)
+            .Select(match => match.Groups[3].Value)
+            .ToList();
+
+        if (!isGitRepo)
+        {
+            Directory.Delete(gitDirectory, true);
```

Comment:
There should be a `try..finally` pattern used here between running `git init` and deleting the `.git` directory to ensure that the directory gets deleted even if there's an exception thrown after calling `git init`.